# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1320257764 25200
#      Wed Nov 02 11:16:04 2011 -0700
# Node ID 2eea4a2e6463d30cf9c2eed7bf02be210d7cfe32
# Parent  4cf7356a99d07d399308b5863da47fae336acd73
NEWS: Update file with more description of changes made to string functions in 3.6.0

* NEWS: Update file with more description of changes made to string functions in 3.6.0

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -31,47 +31,55 @@ Summary of important user-visible change
     They now support nearly all Matlab functionality including:
 
     * ML-compatible whitespace and delimiter defaults
 
     * ML-compatible options: 'whitespace', treatasempty',
     format string repeat count, user-specified comment style, uneven-length
     output arrays, %n and %u conversion specifiers (provisionally)
 
- ** Certain string functions have been modified for greater Matlab compatibility
-    and for 15X greater performance when operating on cell array of strings.
+ ** All .m string functions have been modified for better performance or greater
+    Matlab compatibility.  Performance gains of 15X-30X have been demonstrated.
+    Operations on cell array of strings no longer pay quite as high a penalty
+    as those on 2-D character arrays.
 
     deblank:  Now requires character or cellstr input.
     strtrim:  Now requires character or cellstr input.
               No longer trims nulls ("\0") from string for ML compatibility.
     strmatch: Follows documentation precisely and ignores trailing spaces
               in pattern and in string.  Note that Matlab documents this 
               behavior but the implementation does *not* always follow it.
+    substr:   Now possible to specify a negative LEN option which extracts
+              to within LEN of the end of the string.
+    strtok:   Now accepts cellstr input.
+    base2dec, bin2dec, hex2dec: Now accept cellstr inputs.
+    dec2base, dec2bin, dec2hex: Now accept cellstr inputs.
+    index, rindex: Now accept 2-D character array input.
+    strsplit: Now accepts 2-D character array input.
 
  ** Geometry functions derived from Qhull (convhull, delaunay, voronoi)
     have been revamped.  The options passed to the underlying qhull command
     have been changed for better results or for Matlab compatibility.
 
     convhull : Default options are "Qt" for 2D, 3D, 4D inputs
                Default options are "Qt Qx" for 5D and higher
     delaunay : Default options are "Qt Qbb Qc Qz" for 2D and 3D inputs
                Default options are "Qt Qbb Qc Qx" for 4D and higher
     voronoi  : No default arguments
 
- ** Matlab-compatible preference functions:
+ ** Matlab-compatible preference functions added:
 
-      addpref  getpref  ispref  rmpref  setpref
-
- ** Other miscellaneous new functions:
+    addpref  getpref  ispref  rmpref  setpref
 
-      nthargout
-      iscolumn
-      issrow
-      zscore
+ ** Other new functions added in 3.6.0:
 
+    is_dq_string      python 
+    is_sq_string      zscore
+    nthargout   
+    
  ** Deprecated functions.
 
     The following functions were deprecated in Octave 3.2 and have been
     removed from Octave 3.6.
                                            
       create_set          spcholinv    splu   
       dmult               spcumprod    spmax
       iscommand           spcumsum     spmin
